src/client_creds.rs

```diff
@@ -93,7 +94,7 @@ impl ClientCredsSpotify {
     /// * The read token is expired
     /// * The cached token is disabled in the config
     #[maybe_async]
-    pub async fn read_token_cache(&self) -> ClientResult<Option<Token>> {
+    pub async fn read_token_cache(&self) -> Result<Option<Token>, ReadTokenCacheError> {
```

Comment:
This change can be avoided by making this error part of `ClientError`

Reply:
But then all code that deals in `ClientError`s would now have one more variant to think about even though it can’t actually be created from most code, and similarly any code that calls this function now has to deal with every variant of `ClientError` even though only one actually can be created.

I suppose it would technically fit `ClientError`’s stated purpose to use it here, but the API can be made more strongly typed by not using it I think.

Reply:
Fair enough. My reasoning was that if `ReadTokenCacheError` isn't in `ClientError`, then if someone used to return a `ClientError` will now have to define their own custom error type. But I guess that's not really a huge issue, as most libraries will have their own error anyway.